ide/coqide/dune

```diff
@@ -13,7 +13,7 @@
  (public_name coqidetop.opt)
  (package coqide-server)
  (modules idetop)
- (libraries coq-core.toplevel coqide-server.protocol)
+ (libraries coqide_gui coq-core.toplevel coqide-server.protocol)
  (modes native byte)
  (link_flags -linkall))
```

Comment:
I'm curious, why do you need `-linkall` here? Why not let OCaml link only what's needed? You're probably slowing down your build by a good bit doing this.

Reply:
I have no idea.  Perhaps the author of that line, @ejgallego can explain.

If I didn't have that option then `coqide` wouldn't have linked and loaded the GUI event loop?

Reply:
You could always pull in the modules that initialize the gui event loop in the executable's main manually. E.g:

```
let () =
  let module M = Gui in ()
```

Reply:
I'm not familiar with `let module`.  Does the OCaml doc describe the construct and explain how it works?

(I would love to see the OCaml documentation rewritten.  I think the doc is a barrier to use of OCaml.)